app/src/main/java/com/spascoding/taskycourse/feature_register_screen/domain/use_case/AuthenticateUserUseCase.kt

```diff
@@ -0,0 +1,34 @@
+package com.spascoding.taskycourse.feature_register_screen.domain.use_case
```

Comment:
Here, you're mixing data dependencies (Retrofit API) with your domain layer which is not allowed. The Retrofit calls need to be moved into a repository. Generally, the structure is usually like this:
1. interface AuthRepository in domain
2. class AuthRepositoryImpl: AuthRepository in data
3. AuthRepository interface can now be used across the project to execute data related calls